src/pybind/mgr/cephadm/inventory.py

```diff
@@ -66,6 +67,61 @@ def __init__(
         self.hostname = hostname
 
 
+class DaemonDeployQueue:
+    """
+    Queue of daemons cephadm thinks it should deploy
+    Note this only covers fresh deployments, not redeploy/reconfig/restart
+    """
+
+    def __init__(self) -> None:
+        # mapping of hosts to DaemonPlacement objects that are meant
+        # to be deployed on that host
+        self._daemons: Dict[str, List[Tuple[DaemonPlacement, ServiceSpec]]] = {}
+
+    def add_to_queue(self, slots: List[Tuple[DaemonPlacement, ServiceSpec]]) -> None:
```

Comment:
nit: Queue is in the class name so 'add' should be sufficient (or push or put, etc). But the method name doesn't really need to restate this is a queue.